src/cpu_shader/flatten.rs

```diff
@@ -1,94 +1,51 @@
 // Copyright 2023 the Vello Authors
 // SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
 
+use std::f32::consts::FRAC_1_SQRT_2;
+
 use crate::cpu_dispatch::CpuBinding;
 
-use super::util::{Transform, Vec2, ROBUST_EPSILON};
+use super::{
+    euler::{espc_int_approx, espc_int_inv_approx, CubicParams, EulerParams, EulerSeg},
+    util::{Transform, Vec2, ROBUST_EPSILON},
+};
 use vello_encoding::{
     math::f16_to_f32, BumpAllocators, ConfigUniform, LineSoup, Monoid, PathBbox, PathMonoid,
     PathTag, Style, DRAW_INFO_FLAGS_FILL_RULE_BIT,
 };
 
-fn to_minus_one_quarter(x: f32) -> f32 {
-    // could also be written x.powf(-0.25)
-    x.sqrt().sqrt().recip()
-}
-
-const D: f32 = 0.67;
-fn approx_parabola_integral(x: f32) -> f32 {
-    x * to_minus_one_quarter(1.0 - D + (D * D * D * D + 0.25 * x * x))
-}
-
-const B: f32 = 0.39;
-fn approx_parabola_inv_integral(x: f32) -> f32 {
-    x * (1.0 - B + (B * B + 0.5 * x * x)).sqrt()
-}
-
-#[derive(Clone, Copy, Default)]
-struct SubdivResult {
-    val: f32,
-    a0: f32,
-    a2: f32,
+// TODO: remove this
```

Comment:
Remove uses of `log!`? Or should we leave them in as a no-op?

P.S.: IWBN to have a debug macro that's optionally enabled for this purpose in general.

Reply:
I'm ok leaving them in for the time being, I found it useful while developing to be able to turn it on. I agree that a more sophisticated log story would be nice.